recipes/openimageio/3.x/conanfile.py

```diff
@@ -0,0 +1,303 @@
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 17)
+
+        if is_msvc(self) and is_msvc_static_runtime(self) and self.options.shared:
+            raise ConanInvalidConfiguration(
+                "Building shared library with static runtime is not supported!"
+            )
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        # CMake options
+        tc.variables["CMAKE_DEBUG_POSTFIX"] = ""  # Needed for 2.3.x.x+ versions
+        tc.variables["OIIO_BUILD_TOOLS"] = True
+        tc.variables["OIIO_BUILD_TESTS"] = False
+        tc.variables["BUILD_DOCS"] = False
+        tc.variables["INSTALL_DOCS"] = False
+        tc.variables["INSTALL_FONTS"] = False
+        tc.variables["INSTALL_CMAKE_HELPER"] = False
+        tc.variables["EMBEDPLUGINS"] = True
+        tc.variables["USE_EXTERNAL_PUGIXML"] = True
+        tc.variables["BUILD_MISSING_FMT"] = False
+
+        # Conan is normally not used for testing, so fixing this option to not build the tests
+        tc.variables["BUILD_TESTING"] = False
+
+        tc.variables["USE_DCMTK"] = self.options.with_dicom
+        tc.variables["USE_FFMPEG"] = self.options.with_ffmpeg
+        tc.variables["USE_FIELD3D"] = False
+        tc.variables["USE_FREETYPE"] = self.options.with_freetype
+        tc.variables["USE_GIF"] = self.options.with_giflib
+        tc.variables["USE_HDF5"] = self.options.with_hdf5
+        # Needed for jpeg.imageio plugin, libjpeg/libjpeg-turbo selection still works
+        tc.variables["USE_JPEG"] = True
+        # OIIO CMake files are patched to check USE_* flags to require or not use dependencies
+        tc.variables["USE_JPEGTURBO"] = (self.options.with_libjpeg == "libjpeg-turbo")
+        tc.variables["USE_LIBHEIF"] = self.options.with_libheif
+        tc.variables["USE_LIBJXL"] = self.options.with_libjxl
+        tc.variables["USE_LIBPNG"] = self.options.with_libpng
+        tc.variables["USE_LIBRAW"] = self.options.with_raw
+        tc.variables["USE_LIBWEBP"] = self.options.with_libwebp
+        tc.variables["USE_OPENCV"] = self.options.with_opencv
+        tc.variables["USE_OPENGL"] = False
+        tc.variables["USE_OPENJPEG"] = self.options.with_openjpeg
+        tc.variables["USE_OPENVDB"] = self.options.with_openvdb
+        tc.variables["USE_PTEX"] = self.options.with_ptex
+        tc.variables["USE_PYTHON"] = False
+        tc.variables["USE_QT"] = False
+        tc.variables["USE_TBB"] = self.options.with_tbb
+
+        # Unsupported options
+        tc.variables["USE_NUKE"] = False
+        tc.variables["USE_R3DSDK"] = False
+
+        tc.generate()
+        cd = CMakeDeps(self)
+        cd.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
```

Comment:
This should now be done in the `source()` method after the `get()` call

Reply:
Ok, changed.